### ISSUE TITLE
Strict partial loading

### ISSUE THREAD

**Issue by parisholley** [mikro-orm/mikro-orm#3443]
**Is your feature request related to a problem? Please describe.**
```typescript
const result = await mikro.findOneOrFail(MyEntity, 1, {
  fields: ['field1', 'field2'],
});
```

When using with mongo, it is possible to "select" certain fields and return them, however the return type ends up being `Loaded<Entity, 'field1' | 'field2'>` and still allowing the developer to access unselected fields.

**Describe the solution you'd like**
I'd like to see a new `Selected` type which would effectively be an Omit to make sure no one is accessing unqueried data.

**Comment by B4nan** [comment]
Just a note for anybody who would like to give this a shot - this needs to be about the `Loaded` type, not about changing the return type. Partial loading can be combined with populate hints, and the hints are even inferred (if you dont provide `populate`, its inferred from `fields`).

FYI there was someone few months ago on slack trying to implement this, and ended up with quite breaking solution. We might need to wait for v6 with something like this.

**Comment by B4nan** [comment]
I managed to hack initial PoC for this, and to make the property access fail, we would have to get rid of the intersection with the entity type (`type Loaded<T> = T & { ... }`) which is needed for assignability. Will spend few more hours with this during the week, maybe I will have some ideas to get around it.

What I can do relatively easily is to type such not-selected properties to `never`, which won't fail on access, but can at least fail on misusage.

**Comment by B4nan** [comment]
I think I found a way to make this work with actual property omission. It has its own quirks, but generally should work and should be worth it. The only real downside is that the partial loading will need to work the same as populate hints - as a simple array of relation names (with nested dot path support). I think that's a small feature to sacrifice when you compare it to the added strictness of the loaded result.

Still have to crack the nested partial loading support, as well as fix many TS errors in the codebase caused by the changes, but I feel more confident this will land in v6.

**Comment by B4nan** [comment]
Closing as implemented via #4092, let's test this!

**Comment by parisholley** [comment]
woot! thanks @B4nan, will test out soon. have another 50/day :)
